src/model.py

```diff
@@ -33,9 +35,23 @@ def build_model_and_tokenizer_for(
     ]
 
     logger.info(f"Loading the {model_name} model")
-    model = transformers.AutoModelForCausalLM.from_pretrained(
-        model_name, bad_words_ids=bad_words_ids)
-    model.eval().half().to("cuda")
+    # If loading 6B model in, we need to manually download files and then use the accelerate library.
+    if model_name == "PygmalionAI/pygmalion-6b":
+        from accelerate import init_empty_weights, load_checkpoint_and_dispatch
+        index_file = _download_hf_files(model_name)
+        config = transformers.AutoConfig.from_pretrained(model_name)
+        # Manually add bad words
+        config.bad_words_ids = bad_words_ids
+        
+        with init_empty_weights():
+            model = transformers.AutoModelForCausalLM.from_config(config)
+        model = load_checkpoint_and_dispatch(
+            model, index_file, device_map="auto", no_split_module_classes=["GPTJBlock"])
+        model.eval().half()
+    else:
+        model = transformers.AutoModelForCausalLM.from_pretrained(
+            model_name, bad_words_ids=bad_words_ids)
+        model.eval().half().to("cuda")
```

Comment:
Instead of branching on the model name, how do you feel about branching on whether `accelerate` is available? Basically up top you'd do:

```python3
try:
    import accelerate
    USE_ACCELERATE = True
except ImportError:
    USE_ACCELERATE = False
```

And then here you'd branch based on the value of `USE_ACCELERATE`.

Reply:
All suggestions implemented